src/base/handlers/auth.js

```diff
@@ -0,0 +1,23 @@
+import { getAuthenticatedApiClient, getAuthenticatedUser, ensureAuthenticatedUser } from '../../auth';
+import { getConfig } from '../../config';
+
+export default async function auth(requireAuthenticatedUser, hydrateAuthenticatedUser) {
+  // Get a valid access token for authenticated API access.
+  let authenticatedUser = null;
+
+  if (requireAuthenticatedUser) {
+    authenticatedUser = await ensureAuthenticatedUser();
+  } else {
+    authenticatedUser = await getAuthenticatedUser();
+  }
+
+  if (authenticatedUser !== null && hydrateAuthenticatedUser) {
+    const { username } = authenticatedUser;
+    getAuthenticatedApiClient()
```

Comment:
Is it worth a comment here that this does not use await in order to allow the rest of the app bootstrapping to proceed? (I assume this is the reason)

Reply:
Good call.